### PR TITLE
scheduler: add more hot scheduler comments and replace negative rank

### PR DESCRIPTION
**Pull request by lhy1024**
### What problem does this PR solve?

Issue Number: Ref https://github.com/tikv/pd/issues/5691

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/schedule/schedulers/hot_region.go [hunk 1]
```diff
@@ -1242,7 +1245,7 @@ func (bs *balanceSolver) calcProgressiveRankV1() {
 		// If the first priority is better, the progressiveRank is -3.
```
Comment (okJiang) [inline pkg/schedule/schedulers/hot_region.go:1245]:
```suggestion
		// If the first priority is better, the progressiveRank is 3.
```

File: pkg/schedule/schedulers/hot_region.go [hunk 2]
```diff
@@ -1225,12 +1228,12 @@ func (bs *balanceSolver) isUniformSecondPriority(store *statistics.StoreLoadDeta
...
 // See the comments of `solution.progressiveRank` for more about progressive rank.
-// | ↓ firstPriority \ secondPriority → | isBetter | isNotWorsened | Worsened |
-// |   isBetter                         | -4       | -3            | -1 / 0   |
-// |   isNotWorsened                    | -2       | 1             | 1        |
-// |   Worsened                         | 0        | 1             | 1        |
+// | ↓ firstPriority \ secondPriority → | isBetter | isNotWorsened | Worsened  |
+// |   isBetter                         | 4        | 3             | 1         |
+// |   isNotWorsened                    | 2        | -1            | -1        |
+// |   Worsened                         | 0        | -1            | -1        |
 func (bs *balanceSolver) calcProgressiveRankV1() {
```
Comment (okJiang) [inline pkg/schedule/schedulers/hot_region.go:1235]: How about using variables to represent instead of magic numbers?

Comment (lhy1024) [reply pkg/schedule/schedulers/hot_region.go:1235]: It is the score to use comparing.

Comment (okJiang) [reply pkg/schedule/schedulers/hot_region.go:1235]: Yes, it is. But it is still being used in many places for making judgments, which indeed can be hard to understand in the code, so more comments have to be added to explain it. Like these

https://github.com/tikv/pd/blob/cbba49ab32ddd82d86194abd8cf9c25d1db0c667/pkg/schedule/schedulers/hot_region.go#L1373-L1393

I feel like it's really hard to name, so I'm just suggesting something and not rushing to solve it. :)

File: pkg/schedule/schedulers/hot_region_rank_v1.go [hunk 3]
```diff
@@ -29,9 +29,9 @@ func initRankV1(r *balanceSolver) *rankV1 {
 }
 
 // isAvailable returns the solution is available.
-// The solution should have no revertRegion and progressiveRank < 0.
+// The solution should have no revertRegion and progressiveRank > 0.
 func (*rankV1) isAvailable(s *solution) bool {
-	return s.progressiveRank < 0
+	return s.progressiveRank > 0 && s.revertRegion == nil
```
[the remark this answers is not in the document]
Comment (lhy1024) [reply pkg/schedule/schedulers/hot_region_rank_v1.go:34]: keep consistent with comments

[the remark this answers is not in the document]
Comment (lhy1024) [reply pkg/schedule/schedulers/hot_region_rank_v1.go:34]: However, it seems that v1 does not enable this feature.

File: pkg/statistics/store_hot_peers_infos.go [hunk 4]
```diff
@@ -166,49 +165,37 @@ func summaryStoresLoadByEngine(
...
 			for i := range peerLoadSum {
 				peerLoadSum[i] += peer.GetLoad(i)
 			}
 			hotPeers = append(hotPeers, peer.Clone())
 		}
-		{
-			// Metric for debug.
-			// TODO: pre-allocate gauge metrics
-			ty := "byte-rate-" + rwTy.String() + "-" + kind.String()
-			hotPeerSummary.WithLabelValues(ty, fmt.Sprintf("%v", id)).Set(peerLoadSum[utils.ByteDim])
```
[the remark this answers is not in the document]
Comment (lhy1024) [reply pkg/statistics/store_hot_peers_infos.go:181]: Yeah, we haven't been using this metrics

**Review by rleungx — APPROVED** [review]

File: pkg/schedule/schedulers/hot_region_rank_v1.go [hunk 5]
```diff
@@ -29,9 +29,10 @@ func initRankV1(r *balanceSolver) *rankV1 {
 }
 
 // isAvailable returns the solution is available.
-// The solution should have no revertRegion and progressiveRank < 0.
+// The solution should progressiveRank > 0.
```
Comment (HuSharp) [inline pkg/schedule/schedulers/hot_region_rank_v1.go:32]: Does the comment in `type solution struct` about `progressiveRank` also need to be changed?

Comment (lhy1024) [reply pkg/schedule/schedulers/hot_region_rank_v1.go:32]: good catch

File: pkg/schedule/schedulers/hot_region.go [hunk 6]
```diff
@@ -409,9 +411,9 @@ type solution struct {
 
 	// progressiveRank measures the contribution for balance.
 	// The smaller the rank, the better this solution is.
```
Comment (HuSharp) [inline pkg/schedule/schedulers/hot_region.go:413]:
```suggestion
	// The bigger the rank, the better this solution is.
```

**Review by HuSharp — APPROVED** [review]

File: pkg/schedule/schedulers/hot_region_config.go [hunk 7]
```diff
@@ -38,7 +38,6 @@ import (
 )
 
 const (
-
```
Comment (lhy1024) [inline pkg/schedule/schedulers/hot_region_config.go:41]: Because modify config file, PTAL cc @niubell

**Review by niubell — APPROVED** [review]

**Pull request merged**